compiler/optimizations/gpuTransforms.cpp

```diff
@@ -898,29 +931,44 @@ static void generateGpuAndNonGpuPaths(const GpuizableLoop &gpuLoop,
   // if (chpl_task_getRequestedSubloc() >= 0) {
   //   code to determine number of threads to launch kernel with
   //   call the generated GPU kernel
-  // } else {
+  // } [else] {
   //   run the existing loop on the CPU
   // }
-  Expr* condExpr =
-      new CallExpr(PRIM_GREATEROREQUAL,
-                   new CallExpr(PRIM_GET_REQUESTED_SUBLOC),
-                   new_IntSymbol(0));
-  BlockStmt* thenBlock = new BlockStmt();
-  BlockStmt* elseBlock = new BlockStmt();
-  CondStmt* loopCloneCond = new CondStmt(condExpr, thenBlock, elseBlock);
-  BlockStmt* gpuLaunchBlock = new BlockStmt();
+  //
+  // Normally, We put the CPU block as the else block. If we are not doing GPU
+  // codegen, we put it as an anonymous block right after the conditional. This
+  // will make sure that we call the runtime support as if there's a GPU, yet
+  // still executing the loop always.
 
-  gpuLoop.loop()->insertBefore(loopCloneCond);
-  thenBlock->insertAtHead(gpuLaunchBlock);
-  elseBlock->insertAtHead(gpuLoop.loop()->remove());
+  BlockStmt* cpuBlock = new BlockStmt();
+  BlockStmt* gpuBlock = new BlockStmt();
 
-  VarSymbol *numThreads = generateNumThreads(gpuLaunchBlock, gpuLoop);
+  // populate the gpu block
+  VarSymbol *numThreads = generateNumThreads(gpuBlock, gpuLoop);
   CallExpr* gpuCall = generateGPUCall(kernel, numThreads);
-  gpuLaunchBlock->insertAtTail(gpuCall);
-  gpuLaunchBlock->flattenAndRemove();
+  gpuBlock->insertAtTail(gpuCall);
+
+  CallExpr* condExpr = new CallExpr(PRIM_GREATEROREQUAL,
+                                    new CallExpr(PRIM_GET_REQUESTED_SUBLOC),
+                                    new_IntSymbol(0));
+
+  // we can't add elseStmt later on
+  CondStmt* cond = new CondStmt(condExpr, gpuBlock,
+                                doGpuCodegen() ? cpuBlock : NULL);
+
+  // first, make sure the conditional is in place
+  gpuLoop.loop()->insertBefore(cond);
+
+  // then relocate the loop
+  cpuBlock->insertAtHead(gpuLoop.loop()->remove());
+
+  // if not doing codegen, just add cpuBlock after the conditional
```

Comment:
You need `if not doing GPU codegen` in: `// if not doing codegen, just add cpuBlock after the conditional`